src/openforms/prefill/service.py

```
@@ -0,0 +1,188 @@
+        key_source_mappings[variable.form_variable.key] = variable.form_variable.source
+        # variables which have prefill enabled via the component's configuration
+        if (
+            variable.form_variable.source == FormVariableSources.component
+            and variable.form_variable.prefill_plugin
+            and variable.form_variable.prefill_attribute
+        ):
+            variables_with_attribute.append(variable)
+
+        if variable.form_variable.source == FormVariableSources.user_defined:
+            # variables which have prefill enabled via the variables tab and define prefill options
+            if (
+                variable.form_variable.prefill_plugin
+                and variable.form_variable.prefill_options
+            ):
+                variables_with_options.append(variable)
+
+            # variables which have prefill enabled via the variables tab and define prefill attribute
+            if (
+                variable.form_variable.prefill_plugin
+                and variable.form_variable.prefill_attribute
+            ):
+                variables_with_attribute.append(variable)
+
+    if variables_with_attribute and (
+        results_from_attribute := fetch_prefill_values_from_attribute(
+            submission, register, variables_with_attribute
+        )
+    ):
+        prefill_data.update(**results_from_attribute)
+    if variables_with_options and (
+        results_from_options := fetch_prefill_values_from_options(
+            submission, register, variables_with_options
+        )
+    ):
+        prefill_data.update(**results_from_options)
+
+    total_config_wrapper = submission.total_configuration_wrapper
+    for variable_key, prefill_value in prefill_data.items():
+        if key_source_mappings[variable_key] == FormVariableSources.component:
+            try:
+                component = total_config_wrapper.component_map[variable_key]
+            except KeyError:
+                return None
```

This looks a bit weird - if something is wrong we seem to discard everything. Does that make sense? The lack of coverage on the error handling also makes this seem like a what-if.

This could occur if someone manually creates a `FormVariable` in the admin/shell without it being backed by an actual form definition component, which shouldn't happen. The API endpoint synchronizes the components and derived variables, so I'm leaning towards removing the try-except and if it crashes, it crashes and at least it will become visible in Sentry instead of failing silently, no?

In the old situation this wouldn't happen anyway because we looped only over the prefill variables.